Lib/dataclasses.py

```diff
@@ -1198,6 +1208,22 @@ class C(Base):
     # We use `types.new_class()` instead of simply `type()` to allow dynamic creation
     # of generic dataclassses.
     cls = types.new_class(cls_name, bases, {}, lambda ns: ns.update(namespace))
+
+    # TODO: this hack is the same that can be found in enum.py and should be
+    # removed if there ever is a way to get the caller module.
+    if module is None:
+        try:
+            module = sys._getframe(1).f_globals['__name__']
+        except (AttributeError, ValueError):
+            pass
+    if module is None:
+        _make_class_unpicklable(cls)
+    else:
+        cls.__module__ = module
```

Comment:
Maybe will there any magic to avoid the double `if module is None` ?

Reply:
I'm not sure there is unless nesting deeply.

Reply:
Yes you are right